### ISSUE TITLE
Dialog: ui-state-error is not added on inputtext

### ISSUE THREAD

**Issue by kurtomerfaruk** [primefaces/primefaces#10552]
### Describe the bug

Although I add required property to the inputtext in the dialog, it does not add ui-state-error. I did not have such a problem before PrimeFaces 12 and PrimeFaces 13. How should I follow a path about this. When I use it as a Composite Component, it does not add ui-state-error, but when I use it normally in a dialog, ui-state-error is added

### Reproducer

Used In

PrimeFaces 13.0.0
Jakarta 10 (jakarta.jakartaee-api)
Jakarta Faces (4.0.3)
JDK 11

createpage.xhtml

```xml
<ui:component xmlns:ui="jakarta.faces.facelets"
      xmlns:p="http://primefaces.org/ui"
      xmlns:composite="jakarta.faces.composite"
      xmlns:f="jakarta.faces.core"
      xmlns:h="jakarta.faces.html">

<composite:interface>
    <composite:attribute name="id"/>
    <composite:attribute name="widgetVar"/>
    <composite:attribute name="renderedListener"/>
    <composite:attribute name="actionListener" method-signature="void actionListener(jakarta.faces.event.ActionEvent)" required="true"/>
    <composite:attribute name="kaydetDevamEt" method-signature="void actionListener(jakarta.faces.event.ActionEvent)" required="false"/>
    <composite:attribute name="listForm"/>
    <composite:attribute name="columns" default="2"/>
    <composite:attribute name="focus"/>
    <composite:attribute name="onShow"/>
    <composite:attribute name="position" default="#{null}"/>
</composite:interface>


<composite:implementation>
    <p:dialog widgetVar="#{cc.attrs.widgetVar}"
              modal="true"
              resizable="false"
              closeOnEscape="true"
              header="#{messages.dialogYeniKayit}"
              appendTo="@(body)"
              focus="#{cc.attrs.focus}"
              responsive="true"
              onShow="PF('#{cc.attrs.widgetVar}').initPosition();#{cc.attrs.onShow}"
              fitViewport="true"
              position="#{cc.attrs.position}">

        <h:form id="#{cc.id}">
            <h:panelGroup id="display">
                <p:focus context="display"/>
                <p:panelGrid columns="#{cc.attrs.columns}" columnClasses="column" rendered="#{cc.attrs.renderedListener}" layout="tabular">
                    <composite:insertChildren/>
                    <f:facet name="footer">
                        <div class="float-end">
                            <p:commandButton icon="pi pi-check" styleClass="ui-button-success" actionListener="#{cc.attrs.actionListener}"
                                             value="#{messages.buttonKaydet}" update="display,:#{cc.attrs.listForm}:datalist:datalist,:growl"
                                             oncomplete="handleSubmit(xhr,status,args,PF('#{cc.attrs.widgetVar}'));" onstart="PF('Loading').show()"
                                             onsuccess="PF('Loading').hide()"/>
                            <p:commandButton icon="pi pi-check-square" styleClass="ui-button-help" actionListener="#{cc.attrs.kaydetDevamEt}"
                                             value="#{messages.buttonKaydetDevamEt}" update="display,:#{cc.attrs.listForm}:datalist:datalist,:growl"
                                             onstart="PF('Loading').show()" onsuccess="PF('Loading').hide()"/>
                            <p:commandButton icon="pi pi-times" styleClass="ui-button-danger" value="#{messages.buttonIptal}"
                                             onclick="PF('#{cc.attrs.widgetVar}').hide()" immediate="true" process="@this" resetValues="true"
                                             onstart="PF('Loading').show()" onsuccess="PF('Loading').hide()">
                                    <p:resetInput target="@form"/>
                            </p:commandButton>
                        </div>
                    </f:facet>
                </p:panelGrid>
            </h:panelGroup>
        </h:form>

    </p:dialog>
</composite:implementation>
</ui:component>`
```

Create.xhtml

```xml
<!DOCTYPE html>
<html xmlns:f="jakarta.faces.core"
      xmlns:p="http://primefaces.org/ui"
      xmlns:coco="jakarta.faces.composite/ui">


        <coco:createpage id="AraclarCreateForm"
                         renderedListener="#{araclarController.selected != null}"
                         widgetVar="AraclarCreateDialog"
                         actionListener="#{araclarController.saveNew}"
                         listForm="AraclarListForm"
                         kaydetDevamEt="#{araclarController.kaydetDevamEt}">

            <p:outputLabel value="#{messages.plakaLabel}" for="plaka" />
            <p:inputText id="plaka" value="#{araclarController.selected.plaka}" title="#{messages.plakaText}" required="true" size="45" maxlength="45"/>


            <p:outputLabel value="#{messages.aracCinsiLabel}" for="aracCinsi" />
            <p:inputText id="aracCinsi" value="#{araclarController.selected.aracCinsi}" title="#{messages.aracCinsiText}" required="true" size="45" maxlength="45"/>

            <p:outputLabel value="#{messages.aracTuruLabel}" for="aracTuru" />
            <p:selectOneMenu id="aracTuru" value="#{araclarController.selected.aracTuru}" required="true">
                <f:selectItem itemValue="#{null}" itemLabel="#{messages.seciniz}"/>
                <f:selectItems value="#{araclarController.aracTurus}"
                               var="aracTurus"
                               itemValue="#{aracTurus}"
                               itemLabel="#{messages['aracTuru'+=aracTurus]}"/>
            </p:selectOneMenu>

            <p:outputLabel value="#{messages.personelLabel}" for="personelId" />
            <p:selectOneMenu id="personelId" value="#{araclarController.selected.personelId}" filter="true"  converter="omnifaces.SelectItemsConverter">
                <f:selectItem itemLabel="#{messages.SelectOneMessage}"/>
                <f:selectItems value="#{personellerController.items}"
                               var="personelIdItem"
                               itemValue="#{personelIdItem}"
                               itemLabel="#{personelIdItem.adSoyad.toString()}"/>
            </p:selectOneMenu>
            
            <p:outputLabel value="#{messages.ilceLabel}" for="ilce"/>
            <p:selectOneMenu id="ilce" value="#{araclarController.selected.ilceId}" filter="true" required="true" converter="omnifaces.SelectItemsConverter">
                <f:selectItem itemLabel="#{messages.SelectOneMessage}" itemValue="#{null}"/>
                <f:selectItems value="#{ilcelerController.items}"
                               var="ilceler"
                               itemValue="#{ilceler}"
                               itemLabel="#{ilceler.tanim.toString()}"/>
            </p:selectOneMenu>

        </coco:createpage>

</html>
```

Javascript

```js
function handleSubmit(xhr, status, args, dialog) {
    var dialogId = dialog.id.replaceAll(":","\\\:");
    if (args.validationFailed) {
        $( "#"+dialogId ).effect( "shake", {times: 3}, 100 );
    } else {
        dialog.hide();
    }
}
```

### PrimeFaces edition

Community

### PrimeFaces version

13.0.0

### JSF implementation

Mojarra

### JSF version

4.0.3

### Java version

11

### Browser(s)

Chrome 116.0.5845.97

**Comment by tandraschko** [comment]
We need a primefaces-test project

**Comment by kurtomerfaruk** [comment]
[Here](https://github.com/kurtomerfaruk/PrimeFaces-Test) is the link of the project you requested.

**Comment by melloware** [comment]
@kurtomerfaruk your project doesn't run when I run `mvn clean jetty:run`.  Please use: https://github.com/primefaces/primefaces-test

There is a Jakarta branch which has Mojarra 4.0.3 you can use.  https://github.com/primefaces/primefaces-test/tree/jakarta

**Comment by melloware** [comment]
OK reproducer attached: 
[pf-10552.zip](https://github.com/primefaces/primefaces/files/12441851/pf-10552.zip)

@kurtomerfaruk you said this used to work before PF12 and 13 but I see you are using Jakarta EE10 Faces4.  Are you sure it was primefaces related and not Faces versions?

**Comment by melloware** [comment]
@kurtomerfaruk in your dialog that works you have `update="growl,@form"` which is why the INput shows the issue in your Composite dialog example you only have `update="growl"` so the input is not showing as Required.

As soon as I add `update="growl,@form"`  they both work the same.

**Comment by kurtomerfaruk** [comment]
I have updated for PrimeFaces Test project.

[Project](https://github.com/kurtomerfaruk/PrimeFaces-Test)

**Comment by kurtomerfaruk** [comment]
> @kurtomerfaruk in your dialog that works you have `update="growl,@form"` which is why the INput shows the issue in your Composite dialog example you only have `update="growl"` so the input is not showing as Required.
> 
> As soon as I add `update="growl,@form"` they both work the same.

Yes. As you said, adding @form as update fixed it. But I've been using it this way since PrimeFaces 6. This was how it worked before the version change.

**Comment by melloware** [comment]
Thank you! Let me investigate.

**Comment by melloware** [comment]
OK its a Mojarra bug you should report there: https://github.com/eclipse-ee4j/mojarra/issues It works fine in MyFaces.

`mvn clean jetty:run -Pmojarra40` **FAILS!**
`mvn clean jetty:run -Pmyfaces40` **WORKS**

I think your issue is actually this one: https://github.com/eclipse-ee4j/mojarra/issues/5214

**Comment by melloware** [comment]
@kurtomerfaruk Mojarra team has fixed this in 4.0.5-SNAPSHOT of Mojarra.
